### ISSUE TITLE
Illegal reflective access operation on mac

### ISSUE THREAD

**Issue by lpepy11** [JFormDesigner/FlatLaf#60]
Hi, great work on this look and feel!

I'm hoping we can clean up an illegal reflective access that is being reported on my mac with openjdk 13 -

	
	WARNING: An illegal reflective access operation has occurred
	WARNING: Illegal reflective access by com.formdev.flatlaf.FlatLaf (file:/Users/****/Downloads/flatlaf-demo-0.26.jar) to constructor com.apple.laf.AquaLookAndFeel()
	WARNING: Please consider reporting this to the maintainers of com.formdev.flatlaf.FlatLaf
	WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
	WARNING: All illegal access operations will be denied in a future release
	

Line 197 in FlatLaf.java appears to be the cause (accessing the AquaLookAndFeel constructor):

	String aquaLafClassName = "com.apple.laf.AquaLookAndFeel";
	try {
		base = (BasicLookAndFeel) Class.forName( aquaLafClassName ).newInstance();
	} catch( Exception ex ) {
		LOG.log( Level.SEVERE, "FlatLaf: Failed to initialize base look and feel '" + aquaLafClassName + "'.", ex );
		throw new IllegalStateException();
	}

I had a bit of a play and found this stops the error being reported:

	String aquaLafClassName = UIManager.getSystemLookAndFeelClassName();
	try {
		UIManager.setLookAndFeel(aquaLafClassName);
		base = (BasicLookAndFeel) UIManager.getLookAndFeel();
	} catch( Exception ex ) {
		LOG.log( Level.SEVERE, "FlatLaf: Failed to initialize base look and feel '" + aquaLafClassName + "'.", ex );
		throw new IllegalStateException();
	}

However, I'm no expert in this area and have no idea if this is inefficient or just plain wrong.

Thanks.

**Comment by smileatom** [comment]
This happened to me last night its some kind of nanny error on newInstance.  Dont get me started on what I think about that error.  Anyway.. just commenting to confirm that I also saw this but I generally develop on JDK 8 so Im ignoring this right now.

**Comment by smileatom** [comment]
PR submitted. https://github.com/JFormDesigner/FlatLaf/pull/61

**Comment by DevCharly** [comment]
@lpepy11 thanks for reporting and for suggesting a fix.

`UIManager.setLookAndFeel(aquaLafClassName)` indeed seems to be the only "legal" way to get an instance of `AquaLookAndFeel`, but invoking `UIManager.setLookAndFeel()` from within `UIManager.setLookAndFeel()` is problematic. This invokes `FlatLaf.uninitialize()` from within `FlatLaf.initialize()`. See stack trace:

```
Thread [AWT-EventQueue-0] (Suspended (breakpoint at line 165 in FlatLaf))	
	FlatLightLaf(FlatLaf).uninitialize() line: 165	
	UIManager.setLookAndFeel(LookAndFeel) line: 580	
	UIManager.setLookAndFeel(String) line: 633	
	FlatLightLaf(FlatLaf).getBase() line: 197	
	FlatLightLaf(FlatLaf).initialize() line: 115	
	UIManager.setLookAndFeel(LookAndFeel) line: 586	
	UIManager.setLookAndFeel(String) line: 633	
	DemoPrefs.initLaf(String[]) line: 69	
	FlatTestFrame.create(String[], String) line: 75	
...
```

Also the active LaF remains `Aqua`, but with UI default values from FlatLaf. This is a mess...

Anyway, I'm not happy with the "base" LaF stuff that is currently used to get basic UI defaults. It helped implementing FlatLaf at the beginning, but in the meantime FlatLaf does probably define most UI values itself in the `.properties` files. So the plan is to get rid of the "base" Laf (which will fix this issue) and define missing UI values from Metal/Aqua in FlatLaf.

**Comment by smileatom** [comment]
eeesh

**Comment by lpepy11** [comment]
@smileatom LOL
@DevCharly Thanks for the analysis and explanation.. I suspected I was missing something. Your plan sounds good!

**Comment by smileatom** [comment]
I think I have a fix for this....wait for it.

**Comment by smileatom** [comment]
So newInstance we deprecated due to some hoo-ha about bypassing checked exceptions.  however calling newInstance on a reflected constructor is A-OK. I think we can all sleep better now.

**Comment by smileatom** [comment]
I have for the first time in my life read the doc and followed instructions
https://github.com/JFormDesigner/FlatLaf/pull/61

**Comment by lpepy11** [comment]
@smileatom Strange, I just tested your PR (#61) and still got the access error. I might re-build and re-test to make sure I pulled in your change..

**Comment by smileatom** [comment]
Yeah the problem may be the module system disallowing reflection and have nothing to do with the fact that newInstance is deprecated.  This is the first time I have tried to use JDK 13 and Im running into issues.

**Comment by smileatom** [comment]
Ok so the supported fix to break module encapulation "legally" is add-exports command line option
https://stackoverflow.com/questions/44056405/whats-the-difference-between-add-exports-and-add-opens-in-java-9

Ideally the solution would be this:
--add-exports java.desktop/com.apple.laf=com.formdev.flatlaf on the command line, 

but when I run the test FlatComponentTest I get:
WARNING: Unknown module: com.formdev.flatlaf specified to --add-exports

THIS:
--add-exports java.desktop/com.apple.laf=ALL-UNNAMED
Does actually work, but since its a command line arg the question is how to integrate this

**Comment by smileatom** [comment]
See this doc https://openjdk.java.net/jeps/261

**Comment by smileatom** [comment]
Reopened: https://github.com/JFormDesigner/FlatLaf/pull/61
The fix is to delegate to createLookAndFeel for java 9 and later and use direct reflection on java 8 and earlier.

**Comment by smileatom** [comment]
The issue is not reflection or deprecation but reflecting into a package (aqua L&F) that is not exported from java.desktop due to it being internal.  Java 9 introduced createLookAndFeel which *is* exported and more or less just calls newInstance without initializing the L&F.  It also called using reflection in the PR to compile on jdk 8

**Comment by lpepy11** [comment]
Awesome job @smileatom. I tested your PR (with openjdk 13) and appears to work great. Thanks for your persistence.

I'm happy to leave it up to @DevCharly as to whether he wants to include this fix or re-work as per his original plans.

**Comment by smileatom** [comment]
Im running into other module based errors on JDK 13 but this is a learning experience for me since Ive avoided JDK 9+ runtimes since the day they came out. ;)

**Comment by DevCharly** [comment]
@smileatom many thanks for the fix. This works fine. I've merged it into master branch.

**Comment by DevCharly** [comment]
released in 0.27
